lms/djangoapps/verified_track_content/enrollment_user_tag.py

```diff
@@ -0,0 +1,101 @@
+"""
+User Tag Provider for enrollment tracks.
+"""
+from django.utils.translation import ugettext as _
+
+from course_modes.models import CourseMode
+from openedx.core.djangoapps.users.user_tags.contexts.course import UserTagProviderCourseContext
+from openedx.core.djangoapps.users.user_tags.tag import UserTag
+from student.models import CourseEnrollment
+from verified_track_content.models import VerifiedTrackCohortedCourse
+
+
+# TODO This should really going into common/djangoapps/student/.
+# However, stevedore isn't able to find the student module since
+# it's confused by the common directory magic.
+
+
+class EnrollmentTrackUserTag(UserTag):
+    """
+    UserTag class for enrollment tracks.
+    """
+    def __init__(self, course_mode):
+        self._course_mode = course_mode
+
+    @property
+    def name(self):
+        return self._course_mode.slug
+
+    @property
+    def display_name(self):
+        return unicode(self._course_mode.name)
+
+    @property
+    def description(self):
+        return _(u'Enrollment track for {}').format(self.display_name)
+
+    @property
+    def course_key(self):
+        """
+        Returns the course key associated with this tag.
+        """
+        return self._course_mode.course_id
+
+    @property
+    def users(self):
+        """
+        Returns the users associated with this tag.
+        """
+        return [enrollment.user for enrollment in CourseEnrollment.enrollments_for_mode(self.course_key, self.name)]
+
+
+class EnrollmentTrackUserTagProvider(UserTagProviderCourseContext):
+    """
+    UserTagProvider class for enrollment tracks.
+
+    Note: If VerifiedTrackCohortedCourse is enabled for a course, then
+        tagging is deferred to the CohortUserTagProvider in order to
+        avoid double-tagging based on enrollment tracks.
+    """
+    tag_type = EnrollmentTrackUserTag
+
+    @classmethod
+    def name(cls):
+        return u'openedx.enrollment.track'
+
+    @classmethod
+    def course_tags(cls, course_context, **kwargs):
+        if cls._is_course_using_cohort_instead(course_context):
+            return []
+        course_modes = CourseMode.modes_for_course(course_context.id)
+        return [EnrollmentTrackUserTag(course_mode) for course_mode in course_modes]
```

Comment:
Note that we will now have the potential for more than 2 tags per course that will need to be associated with content. Before it was "verified" and "everything else", but "everything else" may correspond to multiple enrollment modes. This doesn't seem like a big deal, except if additional enrollment modes are added after course teams start associating content with tags; in that case, the newly added enrollment mode would not be associated with any content that was differentiated by track.

It also means that when a course is re-run, it is critical that it have the exact same enrollment modes. And we need to handle the edge case of content being linked to a tag that doesn't actually exist for the course.